kwave/utils/mapgen.py

```diff
@@ -10,13 +10,134 @@
 
 from .conversion import db2neper, neper2db
 from .data import scale_SI
-from .math import cosd, sind, Rz, Ry, Rx
+from .math import cosd, sind, Rz, Ry, Rx, compute_linear_transform
 from .matlab import matlab_assign, matlab_find, ind2sub, sub2ind
 from .matrix import max_nd
 from .tictoc import TicToc
 from ..data import Vector
 
 
+def make_cart_disc(disc_pos: np.ndarray, radius: float, focus_pos: np.ndarray, num_points: int, plot_disc: bool = False,
+                   use_spiral: bool = False) -> np.ndarray:
+    """
+    Create evenly distributed Cartesian points covering a disc.
+
+    Args:
+        disc_pos:       Cartesian position of the center of the disc given as a
+                        two (2D) or three (3D) element tuple [m].
+        radius:         Radius of the disc [m].
+        focus_pos:      Any point on the beam axis of the disc given as a three
+                        element tuple [fx, fy, fz] [m]. Can be set to None to define a disc in the x-y plane.
+        num_points:     Number of points on the disc.
+        plot_disc:      Boolean controlling whether the Cartesian points are plotted (default = False).
+        use_spiral:     Boolean controlling whether the Cartesian points are chosen using a spiral sampling pattern.
+                        Concentric sampling is used by default.
+
+    Returns:
+        disc: 2 x num_points or 3 x num_points array of Cartesian coordinates.
+    """
+    # define literals
+    GOLDEN_ANGLE = 2.39996322972865332223155550663361385312499901105811504
```

Comment:
Consider moving this to global or better to constants file once you merge `make_cart_bowl (...)` PR.